src/main/java/org/mapstruct/intellij/util/MapstructAnnotationUtils.java

```diff
@@ -320,19 +320,15 @@ else if ( mappingsValue instanceof PsiAnnotation ) {
     private static Stream<PsiAnnotation> findMappingAnnotations(@NotNull PsiModifierListOwner method,
                                                                 boolean includeMetaAnnotations) {
 
-        Stream<PsiAnnotation> metaAnnotations = Stream.empty();
-
         if ( includeMetaAnnotations ) {
             // do not use MetaAnnotationUtil#findMetaAnnotations since it only finds the first @Mapping annotation
-            metaAnnotations = findMetaAnnotations( method, new HashSet<>() ).stream();
+            return findMetaAnnotations( method, new HashSet<>() ).stream();
         }
 
-        Stream<PsiAnnotation> directAnnotations = Stream.of( method.getModifierList() )
+        return Stream.of( method.getModifierList() )
             .filter( Objects::nonNull )
             .flatMap( psiModifierList -> Arrays.stream( psiModifierList.getAnnotations() ) )
             .filter( MapstructAnnotationUtils::isMappingAnnotation );
-
-        return Stream.concat( directAnnotations, metaAnnotations );
```

Comment:
Why are this changes needed? I need to double check how we are doing it in the processor. However, meta annotations should be merged with the direct annotations.

e.g.

```java
@Mapping(target = "fromMeta", ignore = true)
public @interface MyCustomMapping {}

@Mapper
public interface MyMapper {

    @MyCustomMapping
    @Mapping(target = "fromMethod", ignore = true)
    Target map(Source source);
}
```

The available annotations for the method is the explicit `@Mapping` on it and all the ones in the `@MyCustomMapping`. Strange, how no tests are failing.

Reply:
This is where the duplicates came from. The method `findMetaAnnotations` finds the direct mappings *and* the meta mappings. Therefore with mapstruct version 1.4+ I'm only calling this to prevent duplicates.

So the merging of direct and meta mappings are getting merged in this method.

Reply:
I now renamed the method to `findDirectAndMetaAnnotations`. Initially I named it only `findMetaAnnotations` because I followed the naming of `MetaAnnotationUtil#findMetaAnnotations` which was not sufficient for us to use.

Reply:
Good point. I didn't analyse the method that well, I was mostly looking at the namings. Good idea for changing the name of the method. Now it correctly shows the intention of it.